### PR TITLE
migrate volume/csi/csi_block.go logs to structured logging

### PR DESCRIPTION
**Pull request by CKchen0726**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
part of kubernetes/enhancements#1602

Which issue(s) this PR fixes:
keps/sig-instrumentation/1602-structured-logging

Structured Logging migration instructions

Special notes for your reviewer:

Does this PR introduce a user-facing change?
```release-note
migrate volume/csi/csi_block.go logs to structured logging
```

### CODE REVIEW

File: pkg/volume/csi/csi_block.go [hunk 1]
```diff
@@ -306,7 +306,7 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 			cleanupErr := m.cleanupOrphanDeviceFiles()
 			if cleanupErr != nil {
 				// V(4) for not so serious error
-				klog.V(4).Infof("Failed to clean up block volume directory %s", cleanupErr)
+				klog.V(4).ErrorS(cleanupErr, "Failed to clean up block volume directory")
```
Comment (lala123912) [inline pkg/volume/csi/csi_block.go:309]: used InfoS instead

File: pkg/volume/csi/csi_block.go [hunk 2]
```diff
@@ -440,7 +440,7 @@ func (m *csiBlockMapper) TearDownDevice(globalMapPath, devicePath string) error
 	}
 	if err = m.cleanupOrphanDeviceFiles(); err != nil {
 		// V(4) for not so serious error
-		klog.V(4).Infof("Failed to clean up block volume directory %s", err)
+		klog.V(4).ErrorS(err, "Failed to clean up block volume directory")
```
Comment (lala123912) [inline pkg/volume/csi/csi_block.go:443]: ErrorS -> InfoS

File: pkg/volume/csi/csi_block.go [hunk 3]
```diff
@@ -226,7 +226,7 @@ func (m *csiBlockMapper) publishVolumeForBlock(
 	if err := os.MkdirAll(publishDir, 0750); err != nil {
 		return "", errors.New(log("blockMapper.publishVolumeForBlock failed to create dir %s:  %v", publishDir, err))
 	}
-	klog.V(4).Info(log("blockMapper.publishVolumeForBlock created directory for publishPath successfully [%s]", publishDir))
+	klog.V(4).InfoS(log("BlockMapper.publishVolumeForBlock successfully created directory for publishPath"), "dir", publishDir)
```
Comment (shubheksha) [inline pkg/volume/csi/csi_block.go:229]:
```suggestion
	klog.V(4).InfoS(log("BlockMapper.publishVolumeForBlock successfully created directory for publishPath"), "path", publishDir)
```

Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

File: pkg/volume/csi/csi_block.go [hunk 4]
```diff
@@ -191,7 +191,7 @@ func (m *csiBlockMapper) stageVolumeForBlock(
 		return "", err
 	}
 
-	klog.V(4).Infof(log("blockMapper.stageVolumeForBlock successfully requested NodeStageVolume [%s]", stagingPath))
+	klog.V(4).InfoS(log("BlockMapper.stageVolumeForBlock successfully requested NodeStageVolume"))
```
Comment (shubheksha) [inline pkg/volume/csi/csi_block.go:194]: The argument for `stagingPath` has been dropped here

File: pkg/volume/csi/csi_block.go [hunk 5]
```diff
@@ -103,7 +103,7 @@ var _ volume.CustomBlockVolumeMapper = &csiBlockMapper{}
 // Example: plugins/kubernetes.io/csi/volumeDevices/{specName}/dev
 func (m *csiBlockMapper) GetGlobalMapPath(spec *volume.Spec) (string, error) {
 	dir := getVolumeDevicePluginDir(m.specName, m.plugin.host)
-	klog.V(4).Infof(log("blockMapper.GetGlobalMapPath = %s", dir))
+	klog.V(4).InfoS(log("BlockMapper.GetGlobalMapPath"), "path", dir)
```
Comment (serathius) [inline pkg/volume/csi/csi_block.go:106]: What does the `log` function? Could you please remove it?

File: pkg/volume/csi/csi_block.go [hunk 6]
```diff
@@ -129,7 +129,7 @@ func (m *csiBlockMapper) getPublishPath() string {
 // returns: pods/{podUID}/volumeDevices/kubernetes.io~csi, {specName}
 func (m *csiBlockMapper) GetPodDeviceMapPath() (string, string) {
 	path := m.plugin.host.GetPodVolumeDeviceDir(m.podUID, utilstrings.EscapeQualifiedName(CSIPluginName))
-	klog.V(4).Infof(log("blockMapper.GetPodDeviceMapPath [path=%s; name=%s]", path, m.specName))
+	klog.V(4).InfoS(log("BlockMapper.GetPodDeviceMapPath"), "path", path, "name", m.specName)
```
Comment (serathius) [inline pkg/volume/csi/csi_block.go:132]: Can you change the key `name` to show what object this name belongs to? Is this a `blockMapperName` a `volumeName` etc

File: pkg/volume/csi/csi_block.go (hunk 4 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_block.go:194]: There is already a log(at line 175) that contains the `stagingPath`, I think it is not necessary to print it again.

File: pkg/volume/csi/csi_block.go (hunk 5 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_block.go:106]: The log() method currently adds "kubernetes.io/csi: " to the beginning of the log message. In my opinion, "kubernetes.io/csi" can be regarded as a label of the log, and there are too many callers, maybe we keep the status quo?

File: pkg/volume/csi/csi_block.go (hunk 2 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_block.go]: updated

File: pkg/volume/csi/csi_block.go (hunk 1 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_block.go]: updated

File: pkg/volume/csi/csi_block.go [hunk 7]
```diff
@@ -141,18 +141,18 @@ func (m *csiBlockMapper) stageVolumeForBlock(
 	csiSource *v1.CSIPersistentVolumeSource,
 	attachment *storage.VolumeAttachment,
 ) (string, error) {
-	klog.V(4).Infof(log("blockMapper.stageVolumeForBlock called"))
+	klog.V(4).InfoS(log("BlockMapper.stageVolumeForBlock called"))
 
 	stagingPath := m.GetStagingPath()
-	klog.V(4).Infof(log("blockMapper.stageVolumeForBlock stagingPath set [%s]", stagingPath))
+	klog.V(4).InfoS(log("BlockMapper.stageVolumeForBlock get stagingPath"), "path", stagingPath)
```
[the remark this answers is not in the document]
Comment (CKchen0726) [reply pkg/volume/csi/csi_block.go:147]: stagingPath indicates a staging path for a directory (on the node) that should be used on NodeStageVolume/NodeUnstageVolume, for example: /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/staging/{specName}

File: pkg/volume/csi/csi_block.go (hunk 6 above)
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_block.go:132]:
```suggestion
	klog.V(4).InfoS(log("BlockMapper.GetPodDeviceMapPath"), "path", path, "volumeName", m.specName)
```

File: pkg/volume/csi/csi_block.go (hunk 5 above)
Comment (serathius) [reply pkg/volume/csi/csi_block.go:106]: We should remove usage of log function, we could consider adding a prefix to all log messages until we propose a better solution (logger naming)

**Comment by serathius** [comment]
Please remove usage of `log` function and then assign issue to me again

**Pull request closed without merging**